R/pkg/R/DataFrame.R

```diff
@@ -392,7 +392,11 @@ setMethod("coltypes",
                 }
 
                 if (is.null(type)) {
-                  stop(paste("Unsupported data type: ", x))
+                  specialtype <- specialtypeshandle(x)
```

Comment:
and then from my comment below, line 1059
`colType <- dtypes[[colIndex]][[2]]`
colType is from dtypes which is the JVM type.
`specialtype <- specialtypeshandle(colType)`
so seem like we are mixing in both cases.